repair_part_source_location/models/stock_move.py

```diff
@@ -0,0 +1,22 @@
+# Copyright 2024 Camptocamp SA (https://www.camptocamp.com).
+# @author: Italo Lopes <italo.lopes@camptocamp.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    def _get_repair_locations(self, repair_line_type, repair_id=False):
+        # This method is used to get the source and destination locations
+        # for the stock move. Will return a tuple with the source and
+        # destination locations.
+        location_id, location_dest_id = super()._get_repair_locations(
+            repair_line_type, repair_id
+        )
+        repair = repair_id or self.repair_id
+        if repair and repair.parts_source_location_id:
+            location_id = repair.parts_source_location_id
+
+        return location_id, location_dest_id
```

Comment:
I'm not sure this is the best way, as it changes the source location for all the moves. 

I recommend changing the value of the `"add"` entry in the `MAP_REPAIR_LINE_TYPE_TO_MOVE_LOCATIONS_FROM_REPAIR` dictionary in the repair module. 

See https://github.com/odoo/odoo/blob/17.0/addons/repair/models/stock_move.py#L7

Reply:
You're right. I missed up the info about change only on ``add`` operation.
Anyway, I think updating this method can be useful in the future, if we need to add some conditions to use the default location or the new field..  What do you think ?
I'll update my PR accordingly.

Reply:
Updated